File: Programming Languages/4 languages in 2 styles/OO/Javascript/Javascript OO/Interfaces/LineFilterer.mjs
export class LineFilterer {
    constructor(wordsToSearchFor) {
        this.wordsToSearchFor = wordsToSearchFor;
    }
    process(fileContent){

    }
}

export class LineFiltererImpl extends LineFilterer {
    constructor(wordsToSearchFor) {
        super(wordsToSearchFor);
    }

    process(fileContent) {
        fileContent = fileContent + '';
        const lines = fileContent.split('\r\n');
        let contentThatMeetsRequirement = "";
        for (let i = 0; i < lines.length; i++) {
            if ((lines[i].toLowerCase().includes(this.wordsToSearchFor.toLowerCase()))) {
                let tempVar = lines[i].substring(0, lines[i].length);
                contentThatMeetsRequirement = contentThatMeetsRequirement + "\r\n" + tempVar;
            }

        }
        return contentThatMeetsRequirement;

    }
}

File: Programming Languages/4 languages in 2 styles/OO/Python/Python OO/LineFilterer.py
class LineFilterer:
    def __init__(self):
        pass


class LineFiltererImpl(LineFilterer):
    def __init__(self, words_to_search_for):
        super().__init__()
        self.words_to_search_for = words_to_search_for

    def process(self, file_content):
        file_content_with_words = ""
        file_content = file_content.split("\n")
        for line in file_content:
            if self.words_to_search_for.lower() in line.lower():
                file_content_with_words += (line + "\n")
        return file_content_with_words
